soc/arm/atmel_sam/common/soc_supc.c

```diff
@@ -30,3 +32,22 @@ void soc_supc_enable_wakeup_source(uint32_t wakeup_source_id)
 
 	SUPC->SUPC_WUMR |= 1 << wakeup_source_id;
 }
+
+void soc_supc_enable_wakeup_pin_source(uint32_t wkup_src,
+				       enum soc_supc_pin_wkup_type wkup_type)
+{
+	uint32_t wuir_high = 1 << (wkup_src + 0x10U);
+
+	SUPC->SUPC_WUIR |= 1 << wkup_src;
```

Comment:
How about `BIT(wkup_src + 0x10U)` and `BIT(wkup_src)`? Below as well, bit of a personal preference I suppose but I think it makes this easier to read, up to you.

Reply:
done